### PR TITLE
Move all commands out of single Cogs

### PR DESCRIPTION
**Pull request by BluePhoenixGame**
*#422 was closed because I reset my changes, can't reopen it :/*

In short, this moves all commands out of their rather empty cogs.
Because simply put it, having one command per cog defeats the purpose of cogs.

Compared to the original way I suggested we do it, this is completely supported by the library!

This will most likely simplify the way you work with the bot, as you can work with the commands directly. Rather than finding them under the cog.

The way git shows the changes... is a little funky, but I just unindented it all mostly...

**Like earlier, this is completely untested**, as there are no set up instructions for how to define the token nor how to set up the database. Also for roles it's not clear how many and what they should be named. I might be missing even more stuff I need to do.

### CODE REVIEW

**Review by MM-coder — APPROVED** [review]
Looks good, I'm not a interpreter though so I'm not sure if there's a hidden bug from all these changes. Approved with reserved confidence in operability

**Review by volcanicer — APPROVED** [review]
LGTM. As MMgamer said there might be a hidden bug somewhere (i.e a `self.x` which shouldn't be there) which isn't surprising given how big this change is, but from what I've seen it looks fine. I'm also glad that an approach that is supported by the library was taken instead of the weird, awkward, unsupported and likely broken implementation in #422.

**Comment by MattW-source** [comment]
I have said on multiple occasions it is set up this way for ease of maintenance and also makes it much easier to disable & enable different functions with !load and !unload (which also allows us to make changes to different parts of the bot without the need of restarting the whole bot.)

I don't see any overall benefit of these changes, and I certainly don't feel comfortable having this within wilfred-370, there are other tasks that need doing for 370 and this isn't one of them, in most cases each command file has 3 to 5 lines removed from them. From a maintenance standpoint, I also see no benefit and if anything this makes it difficult to push small changes without restarting the whole bot. For example if coinflip has a economy breaking bug in it, I can do !unload commands.coinflip which will disable the command without affecting anything else, I can then patch the issue and do !load commands.coinflip which will bring the command back up, with the fix applied.

tl;dr - This adds another layer of complexity and is a huge change to the code structure for little to no benefit.

**Closed**

**Comment by BluePhoenixGame** [comment]
> I have said on multiple occasions it is set up this way for ease of maintenance and also makes it much easier to disable & enable different functions with !load and !unload (which also allows us to make changes to different parts of the bot without the need of restarting the whole bot.)

This still uses extensions, so that shouldn't necessarily change at all.

>  I certainly don't feel comfortable having this within wilfred-370, there are other tasks that need doing for 370 and this isn't one of them

But I have already made most of the changes, left is testing really. This would be an understandable argument if this was an issue and feature request. But I have already done the work and implemented this.
And from a sematical version standpoint, this causes literally 0 breaking changes, and so having this be an introduction in 3.7.0 makes sense.

> From a maintenance standpoint, I also see no benefit and if anything this makes it difficult to push small changes without restarting the whole bot. For example if coinflip has a economy breaking bug in it, I can do !unload commands.coinflip which will disable the command without affecting anything else, I can then patch the issue and do !load commands.coinflip which will bring the command back up, with the fix applied.

Like I said, this still uses extensions and that should be perfectly possible.

> This PR is also way too big to be a stand alone PR, something like this needs an issue associated with it and a target release assigned.

So, the argument is that this is too big to be in one PR? Would you prefer me making one for each command?
I also would argue that this isn't necessarily big, there are no big drastic changes. It's just moving a function out of a class. But because of the repetition for all commands this becomes a lot really.

**Comment by BluePhoenixGame** [comment]
> there might be a hidden bug somewhere (i.e a self.x which shouldn't be there)

I searched the whole repository for the use of self, and fixed it by replacing `self.client` with `ctx.bot` ( self was only ever used for the client attribute).

Though what can be an understandable issue would be if I messed up some of statements with indentation. But I tried to look through the changes, and I don't think I found any unchanged lines ( which means they were unindented ).

**Comment by MattW-source** [comment]
> This still uses extensions, so that shouldn't necessarily change at all.

> Like I said, this still uses extensions and that should be perfectly possible.
I still don't see the benefits

> But I have already made most of the changes, left is testing really. This would be an understandable argument if this was an issue and feature request. But I have already done the work and implemented this.
And from a sematical version standpoint, this causes literally 0 breaking changes, and so having this be an introduction in 3.7.0 makes sense.
That's your fault for doing all the changes without first raising an issue and discussing the need for said changes. It could cause breaking changes due to the sheer volume of code being changed/adjusted, which again would only be clear in testing.

 > So, the argument is that this is too big to be in one PR? Would you prefer me making one for each command?
I also would argue that this isn't necessarily big, there are no big drastic changes. It's just moving a function out of a class. But because of the repetition for all commands this becomes a lot really.
That's not what I meant, I said it's too big to be a **stand alone** PR, this should have been raised as part of an issue/task which is then discussed and assigned to a release, priority and person before changes are made and purposed. This is standard agile procedures, you always make a Change Request/Task before making and purposing changes to be merged.

**Comment by BluePhoenixGame** [comment]
> That's your fault for doing all the changes without first raising an issue and discussing the need for said changes.

I don't care about wasting time. My argument is just that, "you can't say that you need to prioritize other stuff and can't do this, when it's basically already done".

> I still don't see the benefits, these changes carry too much risk for something that I can't see the benefits of doing
>  It could cause breaking changes due to the sheer volume of code being changed/adjusted, which again would only be clear in testing.

Understandable, though the only risk is actual mistakes. Not in any form of redesigning a system.
Like I've said earlier. I want to test this, but I don't know how.

> Had you raised this as an issue/task first and let it be discussed before doing the changes yourself, you wouldn't have risked wasting your time had the request been declined.

Ehh, I don't really care about the time I put into it.

> That's not what I meant, I said it's too big to be a stand alone PR, this should have been raised as part of an issue/task which is then discussed and assigned to a release, priority and person before changes are made and purposed. This is standard agile procedures, you always make a Change Request/Task before making and purposing changes to be merged.

Right, but we're having our discussion in here. The release is 3.7.0, priority doesn't necessarily matter as I do no other things. Otherwise in the sense of the project as a whole, it's quite low. The person would be me!

**Comment by MattW-source** [comment]
> I don't care about wasting time. My argument is just that, "you can't say that you need to prioritize other stuff and can't do this, when it's basically already done".

I can and I will, you are aware that other people who are also making changes as per the tasks in issues will have to work around your changes when it comes to merging. It is likely this would cause merge conflicts due to it changing all files, this creates more work.  

> Right, but we're having our discussion in here. The release is 3.7.0, priority doesn't necessarily matter as I do no other things. Otherwise in the sense of the project as a whole, it's quite low. The person would be me!

You don't decide what goes into what release, the project managers do. Hence why we ask you to raise an issue first so we can allocate things into their respective release cycles. Again, PR's are not where a "discussion" should be held about purposed changes, that is what an issue is for. I am aware other places, including Discord themselves use Pull Requests to discuss changes but it often involves people "reviewing" those changes and also adding in their own. Over here we use the issue system to discuss bugs and change requests. 

> Understandable, though the only risk is actual mistakes. Not in any form of redesigning a system.

Still a risk, and thus more work if a mistake has been made. A redesign of anything would also have to go through a full round of integration tests, to ensure that everything still works with each other as expected, including but not limited to commands working with utilities in helpers.py but also being able to work with each-other, for example !load and !unload would be affected by your changes which hasn't been taken into account in this PR.
